analyses/gistic-vs-focal-cn-file-preparation/01-GISTIC-tidy-data-prep.Rmd

```diff
@@ -0,0 +1,261 @@
+---
+title: "GISTIC tidy data preparation"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: Chante Bethell for ALSF CCDL
+date: 2020
+---
+
+This notebook will tidy the GISTIC data files required to compare the GISTIC's gene level calls with our gene level calls in `analyses/focal-cn-file-prepartion/results`. The required GISTIC files include `all_lesions.conf_90.txt`, `amp_genes.conf_90.txt`, and `del_genes.conf_90.txt` (these file are stored in `analyses/run-gistic/results`). 
+
+The purpose of this notebook is to prepare GISTIC data in a format that can be compared to our focal CN files (both on a gene level and cytoband level). 
+
+## Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/gistic-vs-focal-cn-file-preparation/01-gistic-tidy-data-prep.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+```{r}
+library(tidyverse)
+```
+
+### Files and Directories
+
+```{r}
+# Path to input directory
+gistic_data_dir <- file.path("..", "run-gistic", "results")
+
+# Path to output directory
+results_dir <- "results"
+
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+### Unzip GISTIC results
+
+```{r}
+# Unzip and set up GISTIC results folders and files.
+cohort_gistic_dir <-
+  file.path(gistic_data_dir, "pbta-cnv-consensus-gistic")
+cohort_gistic_zip <-
+  file.path(gistic_data_dir, "pbta-cnv-consensus-gistic.zip")
+
+if (!dir.exists(cohort_gistic_dir)) {
+  unzip(cohort_gistic_zip,
+        exdir = data_dir)
+}
+
+lgat_gistic_dir <-
+  file.path(gistic_data_dir, "pbta-cnv-consensus-lgat-gistic")
+lgat_gistic_zip <-
+  file.path(gistic_data_dir, "pbta-cnv-consensus-lgat-gistic.zip")
+
+if (!dir.exists(lgat_gistic_dir)) {
+  unzip(lgat_gistic_zip,
+        exdir = data_dir)
+}
+```
+
+### Define file paths for GISTIC `all_lesions.conf_90.txt` files
+
+```{r}
+cohort_all_lesions_file <- file.path(cohort_gistic_dir, "all_lesions.conf_90.txt")
+
+lgat_all_lesions_file <- file.path(lgat_gistic_dir, "all_lesions.conf_90.txt")
+```
+
+### Define file path for GISTIC `amp_genes.conf_90.txt` and `del_genes.conf_90.txt` result files
+
+```{r}
+cohort_amp_genes_file <- file.path(cohort_gistic_dir, "amp_genes.conf_90.txt")
+
+lgat_amp_genes_file <- file.path(lgat_gistic_dir, "amp_genes.conf_90.txt")
+
+cohort_del_genes_file <- file.path(cohort_gistic_dir, "del_genes.conf_90.txt")
+
+lgat_del_genes_file <- file.path(lgat_gistic_dir, "del_genes.conf_90.txt")
+```
+
+## Functions to tidy data
+
+```{r}
+# Code adapted from `analyses/gistic-cohort-vs-histology-comparison/gistic-cohort-vs-histology-comparison.Rmd`
+get_genes_df <- function(genes_df) {
```

Comment:
Probably a good indication that it is time for this to get broken out into an R file in a `util`... for the parts that you're adding/changing from `analyses/gistic-cohort-vs-histology-comparison/gistic-cohort-vs-histology-comparison.Rmd` can you write a second function that uses the output from `get_genes_vector` to create a data.frame or is there some other way to make this more generalizable? I think this is also a good indication that this notebook should be in the same module as what is in `gistic-cohort-vs-histology-comparison` but perhaps with a more general name like `compare-gistic`.